### ISSUE TITLE
Vagrant using fedora is not working

### ISSUE THREAD

**Issue by derekwaynecarr** [kubernetes/kubernetes#26104]
Investgating...

**Comment by derekwaynecarr** [comment]
There is an error when attempting to update

```
dnf update -y dnf dnf-plugins-core
```

The update fails for some reason.

Related: https://github.com/saltstack/salt/issues/31001

**Comment by derekwaynecarr** [comment]
```
[vagrant@kubernetes-master ~]$ sudo dnf update -y dnf dnf-plugins-core
Last metadata expiration check performed 0:11:05 ago on Mon May 23 20:45:41 2016.
Dependencies resolved.
===================================================================================================================================================================================================================
 Package                                                       Arch                                        Version                                              Repository                                    Size
===================================================================================================================================================================================================================
Upgrading:
 dnf                                                           noarch                                      1.1.9-1.fc23                                         updates                                      276 k
 dnf-conf                                                      noarch                                      1.1.9-1.fc23                                         updates                                       96 k
 dnf-plugins-core                                              noarch                                      0.1.21-1.fc23                                        updates                                       40 k
 dnf-yum                                                       noarch                                      1.1.9-1.fc23                                         updates                                       79 k
 python2-dnf                                                   noarch                                      1.1.9-1.fc23                                         updates                                      444 k
 python3-dnf                                                   noarch                                      1.1.9-1.fc23                                         updates                                      448 k
 python3-dnf-plugins-core                                      noarch                                      0.1.21-1.fc23                                        updates                                       93 k

Transaction Summary
===================================================================================================================================================================================================================
Upgrade  7 Packages

Total size: 1.4 M
Downloading Packages:
[SKIPPED] dnf-1.1.9-1.fc23.noarch.rpm: Already downloaded                                                                                                                                                         
[SKIPPED] python3-dnf-1.1.9-1.fc23.noarch.rpm: Already downloaded                                                                                                                                                 
[SKIPPED] dnf-conf-1.1.9-1.fc23.noarch.rpm: Already downloaded                                                                                                                                                    
[SKIPPED] dnf-yum-1.1.9-1.fc23.noarch.rpm: Already downloaded                                                                                                                                                     
[SKIPPED] python2-dnf-1.1.9-1.fc23.noarch.rpm: Already downloaded                                                                                                                                                 
[SKIPPED] dnf-plugins-core-0.1.21-1.fc23.noarch.rpm: Already downloaded                                                                                                                                           
[SKIPPED] python3-dnf-plugins-core-0.1.21-1.fc23.noarch.rpm: Already downloaded                                                                                                                                   
Running transaction check
Transaction check succeeded.
Running transaction test
The downloaded packages were saved in cache till the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: Transaction check error:
  file /usr/share/locale/ca/LC_MESSAGES/dnf-plugins-core.mo from install of python3-dnf-plugins-core-0.1.21-1.fc23.noarch conflicts with file from package python-dnf-plugins-core-0.1.13-1.fc23.noarch
  file /usr/share/locale/da/LC_MESSAGES/dnf-plugins-core.mo from install of python3-dnf-plugins-core-0.1.21-1.fc23.noarch conflicts with file from package python-dnf-plugins-core-0.1.13-1.fc23.noarch

Error Summary
```

**Comment by nickcarenza** [comment]
Getting a similar error.

```
          ID: bridge-utils
    Function: pkg.installed
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/site-packages/salt/state.py", line 1564, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/site-packages/salt/states/pkg.py", line 872, in installed
                  **kwargs)
                File "/usr/lib/python2.7/site-packages/salt/states/pkg.py", line 243, in _find_install_targets
                  cur_pkgs = __salt__['pkg.list_pkgs'](versions_as_list=True, **kwargs)
                File "/usr/lib/python2.7/site-packages/salt/modules/yumpkg.py", line 723, in list_pkgs
                  for pkginfo in _repoquery_pkginfo(list_cmd):
                File "/usr/lib/python2.7/site-packages/salt/modules/yumpkg.py", line 134, in _repoquery_pkginfo
                  for line in _repoquery(repoquery_args):
                File "/usr/lib/python2.7/site-packages/salt/modules/yumpkg.py", line 228, in _repoquery
                  _check_repoquery()
                File "/usr/lib/python2.7/site-packages/salt/modules/yumpkg.py", line 190, in _check_repoquery
                  raise CommandExecutionError('Unable to install dnf-plugins-core')
              CommandExecutionError: Unable to install dnf-plugins-core
     Started: 21:18:22.525032
    Duration: 1255.517 ms
     Changes:
```

except that my dnf update reports success

```
[vagrant@kubernetes-master ~]$ sudo dnf update -y dnf dnf-plugins-core
Last metadata expiration check: 1:19:42 ago on Mon May 23 20:02:27 2016.
Dependencies resolved.
Nothing to do.
Complete!
```

**Comment by derekwaynecarr** [comment]
Yeah, so the root error is when salt is installed and begins to provision:

```
==> master: Running transaction
==> master:   Installing  : hwdata-0.289-1.fc23.noarch                                  1/6 
==> master:   Installing  : python2-systemd-231-4.fc23.x86_64                           2/6 
==> master:   Installing  : pciutils-libs-3.3.1-2.fc23.x86_64                           3/6 
==> master:   Installing  : pciutils-3.3.1-2.fc23.x86_64                                4/6 
==> master:   Installing  : salt-2015.5.9-2.fc23.noarch                                 5/6 
==> master:   Installing  : salt-minion-2015.5.9-2.fc23.noarch                          6/6 
==> master:   Verifying   : salt-minion-2015.5.9-2.fc23.noarch                          1/6 
==> master:   Verifying   : salt-2015.5.9-2.fc23.noarch                                 2/6 
==> master:   Verifying   : pciutils-3.3.1-2.fc23.x86_64                                3/6 
==> master:   Verifying   : pciutils-libs-3.3.1-2.fc23.x86_64                           4/6 
==> master:   Verifying   : python2-systemd-231-4.fc23.x86_64                           5/6 
==> master:   Verifying   : hwdata-0.289-1.fc23.noarch                                  6/6 
==> master: 
==> master: Installed:
==> master:   hwdata.noarch 0.289-1.fc23             pciutils.x86_64 3.3.1-2.fc23           
==> master:   pciutils-libs.x86_64 3.3.1-2.fc23      python2-systemd.x86_64 231-4.fc23      
==> master:   salt.noarch 2015.5.9-2.fc23            salt-minion.noarch 2015.5.9-2.fc23     
==> master: 
==> master: Complete!
==> master:  *  INFO: Running install_fedora_stable_post()
==> master: disabled
==> master: Created symlink from /etc/systemd/system/multi-user.target.wants/salt-minion.service to /usr/lib/systemd/system/salt-minion.service.
==> master:  *  INFO: Running install_fedora_check_services()
==> master:  *  INFO: Running install_fedora_restart_daemons()
==> master:  *  INFO: Running daemons_running()
==> master:  *  INFO: Salt installed!
==> master:   Now waiting for the Salt provisioning process to complete on this machine.
==> master:   This can take some time based on your network, disk, and cpu speed.
==> master: [ERROR   ] Command ['dnf', '-y', 'install', 'dnf-plugins-core', 'python-dnf-plugins-core'] failed with return code: 1
==> master: [ERROR   ] output: Last metadata expiration check: 0:02:49 ago on Mon May 23 21:19:07 2016.
==> master: Package dnf-plugins-core-0.1.21-1.fc23.noarch is already installed, skipping.
==> master: Dependencies resolved.
==> master: ================================================================================
==> master:  Package                      Arch        Version            Repository    Size
==> master: ================================================================================
==> master: Installing:
==> master:  python-dnf-plugins-core      noarch      0.1.13-1.fc23      fedora        85 k
==> master:  python-hawkey                x86_64      0.6.2-1.fc23       fedora        77 k
==> master:  python-libcomps              x86_64      0.1.7-1.fc23       fedora        47 k
==> master:  python-librepo               x86_64      1.7.16-2.fc23      fedora        53 k
==> master:  python2-dnf                  noarch      1.1.9-1.fc23       updates      444 k
==> master: 
==> master: Transaction Summary
==> master: ================================================================================
==> master: Install  5 Packages
==> master: 
==> master: Total download size: 707 k
==> master: Installed size: 2.5 M
==> master: Downloading Packages:
==> master: --------------------------------------------------------------------------------
==> master: Total                                           417 kB/s | 707 kB     00:01     
==> master: Running transaction check
==> master: Transaction check succeeded.
==> master: Running transaction test
==> master: The downloaded packages were saved in cache until the next successful transaction.
==> master: You can remove cached packages by executing 'dnf clean packages'.
==> master: Error: Transaction check error:
==> master:   file /usr/share/locale/ca/LC_MESSAGES/dnf-plugins-core.mo from install of python-dnf-plugins-core-0.1.13-1.fc23.noarch conflicts with file from package python3-dnf-plugins-core-0.1.21-1.fc23.noarch
==> master:   file /usr/share/locale/da/LC_MESSAGES/dnf-plugins-core.mo from install of python-dnf-plugins-core-0.1.13-1.fc23.noarch conflicts with file from package python3-dnf-plugins-core-0.1.21-1.fc23.noarch
==> master: 
==> master: Error Summary
==> master: -------------
==> master: [ERROR   ] An exception occurred in this state: Traceback (most recent call last):
==> master:   File "/usr/lib/python2.7/site-packages/salt/state.py", line 1564, in call
==> master:     **cdata['kwargs'])
==> master:   File "/usr/lib/python2.7/site-packages/salt/states/pkg.py", line 872, in installed
==> master:     **kwargs)
==> master:   File "/usr/lib/python2.7/site-packages/salt/states/pkg.py", line 243, in _find_install_targets
==> master:     cur_pkgs = __salt__['pkg.list_pkgs'](versions_as_list=True, **kwargs)
==> master:   File "/usr/lib/python2.7/site-packages/salt/modules/yumpkg.py", line 723, in list_pkgs
==> master:     for pkginfo in _repoquery_pkginfo(list_cmd):
==> master:   File "/usr/lib/python2.7/site-packages/salt/modules/yumpkg.py", line 134, in _repoquery_pkginfo
==> master:     for line in _repoquery(repoquery_args):
==> master:   File "/usr/lib/python2.7/site-packages/salt/modules/yumpkg.py", line 228, in _repoquery
==> master:     _check_repoquery()
==> master:   File "/usr/lib/python2.7/site-packages/salt/modules/yumpkg.py", line 190, in _check_repoquery
==> master:     raise CommandExecutionError('Unable to install dnf-plugins-core')
==> master: CommandExecutionError: Unable to install dnf-plugins-core
```

**Comment by derekwaynecarr** [comment]
The salt repo implied 0.16 worked, but that is no longer available in fedora repos...

Problem is `python-dnf-plugins-core` is not installing, and then problems cascade from there...

**Comment by nickcarenza** [comment]
Looks like there is a conflict:

```
[vagrant@kubernetes-master ~]$ sudo dnf install python-dnf-plugins-core
Last metadata expiration check: 1:41:23 ago on Mon May 23 20:02:27 2016.
Dependencies resolved.
=======================================================================================================================
 Package                               Arch                 Version                        Repository             Size
=======================================================================================================================
Installing:
 python-dnf-plugins-core               noarch               0.1.13-1.fc23                  fedora                 85 k
 python-hawkey                         x86_64               0.6.2-1.fc23                   fedora                 77 k
 python-libcomps                       x86_64               0.1.7-1.fc23                   fedora                 47 k
 python-librepo                        x86_64               1.7.16-2.fc23                  fedora                 53 k
 python2-dnf                           noarch               1.1.9-1.fc23                   updates               444 k

Transaction Summary
=======================================================================================================================
Install  5 Packages

Total size: 707 k
Installed size: 2.5 M
Is this ok [y/N]: y
Downloading Packages:
[SKIPPED] python-dnf-plugins-core-0.1.13-1.fc23.noarch.rpm: Already downloaded
[SKIPPED] python2-dnf-1.1.9-1.fc23.noarch.rpm: Already downloaded
[SKIPPED] python-libcomps-0.1.7-1.fc23.x86_64.rpm: Already downloaded
[SKIPPED] python-librepo-1.7.16-2.fc23.x86_64.rpm: Already downloaded
[SKIPPED] python-hawkey-0.6.2-1.fc23.x86_64.rpm: Already downloaded
Running transaction check
Transaction check succeeded.
Running transaction test
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: Transaction check error:
  file /usr/share/locale/ca/LC_MESSAGES/dnf-plugins-core.mo from install of python-dnf-plugins-core-0.1.13-1.fc23.noarch conflicts with file from package python3-dnf-plugins-core-0.1.21-1.fc23.noarch
  file /usr/share/locale/da/LC_MESSAGES/dnf-plugins-core.mo from install of python-dnf-plugins-core-0.1.13-1.fc23.noarch conflicts with file from package python3-dnf-plugins-core-0.1.21-1.fc23.noarch

Error Summary
-------------

```

Salt seems to require python3-dnf-plugins-core which prevents python-dnf-plugins core from being installed.

**Comment by nickcarenza** [comment]
Solved the issue for me by editing `./cluster/vagrant/provision-utils.sh` with `yum --enablerepo=updates-testing install -y salt-minion` right before `echo "Running release install script"` [here](https://github.com/kubernetes/kubernetes/blob/master/cluster/vagrant/provision-utils.sh#L106). This solution has not been thoroughly tested by any means. It has just allowed `./cluster/kube-up.sh` to run successfully with no failed states in salt provision.

Will update shortly.

Reference: https://docs.saltstack.com/en/latest/topics/installation/fedora.html

**Comment by derekwaynecarr** [comment]
Will try the same.  Thanks for the help.  Hope to have a PR tomorrow morning.

**Comment by derekwaynecarr** [comment]
Posted a fix here: https://github.com/kubernetes/kubernetes/pull/26166

**Comment by vectorsigma** [comment]
FWIW, it's not just with Vagrant.  Regular Fedora 23 on bare metal is affected.

The problem appears to be that the package `python-dnf-plugins-core` is no longer used in Fedora, but Salt still tries to install it.  Unfortunately, that package is still available in the repos (it's part of the base Fedora 23 repo) so Saltstack can still install it by name, as it tries to do.

From my initial investigation, this conundrum appears to be because in Fedora-land, `python3-dnf-plugins-core` obsoletes `python2-dnf-plugins-core` and `python-dnf-plugins-core`, but the latter isn't marked as such.

I'm not quite sure how to resolve this.  Enabling Rawhide just to get highstate to work seems silly.  Which I'm not even sure how it would work, since that particular package is missing.. so perhaps someone added an `obsoletes: python-dnf-plugins-core`, like they did for `python2-dnf-plugins-core` in the `python3-dnf-plugins-core` package?

Anyway, it's bed time.  Just thought I'd note what I found.

**Comment by derekwaynecarr** [comment]
This is now fixed on HEAD as of #26174
